### PR TITLE
braa: init at 0.82

### PR DESCRIPTION
**Pull request by bycEEE**
## Description of changes

Add braa (mass snmp scanner).

URLs:
- https://www.kali.org/tools/braa/
- http://s-tech.elsat.net.pl/

Notes:
I'm very new to the nix programming language and NixOS. Let me know if there's any changes or improvements to make. Additionally, I don't believe I'll be a very active maintainer. Should I still keep my name in the `maintainers` block?

Related to https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/br/braa/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  zlib,
+}:
+stdenv.mkDerivation rec {
+  pname = "braa";
+  version = "0.82";
+
+  src = fetchurl {
+    url = "http://s-tech.elsat.net.pl/${pname}/${pname}-${version}.tar.gz";
+    sha256 = "d650d34cdf30fcbbbbf4e4f11c8f5d51fd5caa3cca58d620fa1d978b8c18919c";
+  };
```
Comment (h7x4) [inline pkgs/by-name/br/braa/package.nix:14]: I believe the correct fetcher here would be `fetchzip`

```suggestion
  src = fetchzip {
    url = "http://s-tech.elsat.net.pl/${pname}/${pname}-${version}.tar.gz";
    hash = "d650d34cdf30fcbbbbf4e4f11c8f5d51fd5caa3cca58d620fa1d978b8c18919c";
  };
```

File: pkgs/by-name/br/braa/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  buildInputs = [zlib];
+
+  installPhase = ''
+    install -Dm755 braa $out/bin/braa
+  '';
```
Comment (h7x4) [inline pkgs/by-name/br/braa/package.nix:20]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 braa $out/bin/braa
    runHook postInstall
  '';
```

File: pkgs/by-name/br/braa/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "A mass snmp scanner.";
```
Comment (h7x4) [inline pkgs/by-name/br/braa/package.nix:23]:
```suggestion
    description = "A mass snmp scanner";
```

**Review by h7x4 — COMMENTED** [review]
Hello @bycEEE, welcome to nixpkgs!

**Comment by h7x4** [comment]
I think a package needs to have a maintainer to be accepted in the first place

File: pkgs/by-name/br/braa/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    homepage = "http://s-tech.elsat.net.pl";
+    license = licenses.gpl2;
```
Comment (eclairevoyant) [inline pkgs/by-name/br/braa/package.nix:25]: Deprecated identifier, please use gpl2Only or gpl2Plus as per upstream

File: pkgs/by-name/br/braa/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    platforms = platforms.unix;
+    maintainers = with maintainers; [bycEEE];
```
Comment (eclairevoyant) [inline pkgs/by-name/br/braa/package.nix:27]: please set `meta.mainProgram` if there is a main binary

File: pkgs/by-name/br/braa/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/br/braa/package.nix:14]: @h7x4 that hash would be incorrect regardless fyi

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by eclairevoyant** [comment]
Thanks for the contribution! Please ensure to move the maintainer commit first, along with the other feedback

**Comment by bycEEE** [comment]
Changes made and commits squashed/re-ordered. I think I did this correctly.

**Review by eclairevoyant — APPROVED** [review]
approving, but FYI you might get some formatting nits from other reviewers, even though I don't think it's relevant, and should be handled by CI if anything. 

Either way just a heads up that using `nixpkgs-fmt` will _usually_ get you closer to the official style than `alejandra` which I assume you used here.

see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax for info

**Review by IogaMaster — APPROVED** [review]

**Pull request merged**